Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/EventTag/ByEventTag.tsx

```diff
@@ -0,0 +1,166 @@
+﻿﻿//******************************************************************************************************
+//  ByEventTag.tsx - Gbtc
+//
+//  Copyright © 2023, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/31/2023 - Parker Dinsdale
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { EventTagSlice } from '../Store/Store';
+
+import Table from '@gpa-gemstone/react-table'
+import { OpenXDA, Application } from '@gpa-gemstone/application-typings';
+import { SearchBar, Search, Modal } from '@gpa-gemstone/react-interactive';
+import { CrossMark, HeavyCheckMark } from '@gpa-gemstone/gpa-symbols';
+import EventTagForm from './EventTagForm';
+
+
+const EventTags: Application.Types.iByComponent = (props) => {
+    const dispatch = useAppDispatch();
+
+    const data = useAppSelector(EventTagSlice.SearchResults);
+    const status = useAppSelector(EventTagSlice.SearchStatus);
+
+    const allTags = useAppSelector(EventTagSlice.Data);
+    const allTagsStatus = useAppSelector(EventTagSlice.Status);
+
+    const [mode, setMode] = React.useState<'View'|'Add'|'Edit'>('View');
+    const [sortField, setSortField] = React.useState<keyof OpenXDA.Types.EventTag>('Name');
```

Comment:
do we really need these as states?
If you are using a `GenericSlice` that already has a `sortField` and `ascending` export,... This may cause issues since these states may not be synced up with what is happening in the Slice